tests/framework/utils.py

```diff
@@ -127,3 +127,23 @@ def shell_subprocess_check_output(command):
     output = subprocess.check_output(command, shell=True)
     logger.info("Output: %s", output)
     return output
+
+def get_pod(namespace, label_selector):
+    """Gets pods in the `namespace`. Returns the first pod that has `label_filter`.
+    Returns None if the number of matches is not equal to 1.
+    """
+    pods = K8S_CLUSTER_MANAGER.k8s_client_dict[CONST.K8S_V1_CLIENT_KEY].list_namespaced_pod(
+            namespace = namespace, label_selector = label_selector
+        )
+    if len(pods.items) != 1:
```

Comment:
Should we handle the cases ==0 and >= 2 separately? (Otherwise the response "no match" might be misleading). Or if >=2 is supposed to be impossible, maybe we should have a warning or exception in that case.

Reply:
Updated the log message.